gnovm/pkg/gnolang/go2gno.go

```diff
@@ -35,12 +35,16 @@ import (
 	"go/ast"
 	"go/parser"
 	"go/token"
+	"go/types"
 	"os"
 	"reflect"
 	"strconv"
+	"strings"
 
 	"github.com/davecgh/go-spew/spew"
 	"github.com/gnolang/gno/tm2/pkg/errors"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"go.uber.org/multierr"
```

Comment:
Just curious: have you considered to use the standard go `errors.Join` instead of this package?
For most use cases, I've found it to be adequate

Reply:
let me know what you think, but I'm kind of the opinion that `multierr` here is somewhat less annoying. I think we can make a separate PR where we change all of `multierr`'s usages into `errors.Join`, but essentially I see two big annoyances:

1. the `errors` package clashes with the `tm2/pkg/errors`; so any imports have to be named with something like `stderrors`.
2. I don't want to look up the discussions on the Go issue tracker, but for some reason there is no matching `Unwrap` function for the `Unwrap() []error` method. What I mean: multierr has the nice `multierr.Errors` function, which we use to create the resulting error message in `sdk/vm`. Go's standard `errors` does _not_. Most Go updates that folks disagree with don't bother me; this one really does.

Essentially, what I'd like is to have `pkg/errors` add the functions `Join`, as an alias for Go's std `errors.Join`, so that the naming mess is "solved"; and an `Errors` method which does the unwrapping on the joined error.

Reply:
No need to swap anything in this PR, was just curious :)

I agree with the points 💯